tfhe/src/integer/server_key/radix_parallel/scalar_sub.rs

```diff
@@ -177,7 +177,10 @@ impl ServerKey {
             // And here, it's because shortint sub_assign adds a correcting term,
             // which we do not want here
             crate::core_crypto::algorithms::lwe_ciphertext_sub_assign(&mut lhs_b.ct, &borrow.ct);
-            lhs_b.set_noise_level(lhs_b.noise_level() + borrow.noise_level());
+            lhs_b.set_noise_level(
+                lhs_b.noise_level() + borrow.noise_level(),
+                self.key.max_noise_level,
```

Comment:
`ServerKey.max_degree` is initialized with `MaxDegree::integer_radix_server_key` in radix integers
This keeps space for carry propagation in all operations (except in carry propagation I guess, not sure if there are degree checks there)

I think we should do something similar for noise level
Or are noise checks in carry propagation enough? If so, I guess it would mean some operations could panic in carry propagation even if we could have prevented it in the previous operation (adding a PBS if the `max_noise_level_keep_space_for_carry_propagation` is too big)  

In practice not sure this is a problem as noise and degree checks are often redundant.

Reply:
While I do thing max_degree and max_noise_level should be aligned I don't know if its better to align max_degree to max_noise_level or the reverse

Reply:
The only use of the ServerKey.max_degree having space for 1 carry addition is for smart_add.sub/neg to work if the carry_propagation algorithm is the sequential one, not parallel

All other algorithm don't rely on this

Reply:
max_noise_level is separate from the max degree and depends on the optimization of parameters

Reply:
I think it's the question about "what do the parameters allow ?" i.e. the max_max_degree and the max_max_noise_level

and

"what are my constraints for these integers" which are the max_degree and max_noise_level

dont know if this is the right PR for this change (having both notions)

Reply:
maybe NoiseLevel::MAX makes no sense to be honest